### ISSUE TITLE
fade.out truncates file

### ISSUE THREAD

**Issue by ohrstrom** [savonet/liquidsoap#1159]
Dear liquidsoap crew

@toots - we did chat about this a couple of weeks ago, you did [ask me to file an issue](https://liquidsoapworkspace.slack.com/archives/CJWSG659U/p1582841489026300) - so here you are :)

**Describe the bug**
Applying `fade.out` to a source results in unexpected behaviour:  
The last second(s) of an audio-file passed to `fade.out` get's cut.

**To Reproduce**

```sh
liquidsoap \
    'set("log.stdout",true)
    set("log.level",4)
    set("server.telnet",true)

    s = audio_to_stereo(request.queue(id="s0"))
    s = fade.out(duration=0.0,s)
    s = mksafe(s)

    output.pulseaudio(
        id="output_pulseaudio",
        fallible=false,
        s
    )'
```

Example file
```
wget -qO- https://github.com/savonet/liquidsoap/files/4480679/123.mp3.zip | tar -xf-
```

**Expected behaviour**

1. Sending a "play request" to ls without any annotations should (i.m.o.) play the file until it's end:
```sh
echo "s0.push ./123.mp3" | nc 127.0.0.1 1234
```
2. Annotating to fade out `0` seconds should play the file until it's end:
```sh
echo "s0.push annotate:liq_fade_out="0.0":./123.mp3" | nc 127.0.0.1 1234 
```
3. Annotating to fade out `0.1` seconds should fade out during the last 0.1 seconds:
```sh
echo "s0.push annotate:liq_fade_out="0.1":./123.mp3" | nc 127.0.0.1 1234 
```
4. Annotating to fade out `1` second should fade out during the last 1 second:
```sh
echo "s0.push annotate:liq_fade_out="1.0":./123.mp3" | nc 127.0.0.1 1234 
```

**Actual behaviour**

1. The last + / - one second get's cut.

2. The last + / - one second get's cut.

3. The last + / - one second get's cut.

4. Works as expected.

**Version details**
 - OS: same behaviour for Debian and MacOs
 - Version `1.4.0` & `1.4.1` (opam)

[123.mp3.zip](https://github.com/savonet/liquidsoap/files/4480679/123.mp3.zip)

**Comment by toots** [comment]
Thanks for a truly awesome bug report! I was able to locate the issue and fix it. It should be fixed in both `master` and our upcoming `1.4.2` release.

Any chance you could test the `1.4.2` branch? All you should have to do is:
```
git clone --recursive https://github.com/savonet/liquidsoap.git
cd liquidsoap && git checkout 1.4.2
opam install -y .
```

**Comment by ohrstrom** [comment]
Hi @toots - wow thanks a lot!! I did just install `1.4.2` and it is working perfectly!
